src/command.rs

```diff
@@ -139,3 +167,89 @@ impl ShellCommand for ExternalCommand {
         Ok(child.stdout.take())
     }
 }
+
+#[derive(Debug, Clone)]
+pub struct LlmCommand {
+    prompt: String,
+    openai_client: OpenAIClient,
+}
+
+impl LlmCommand {
+    pub fn new(prompt: String, openai_client: OpenAIClient) -> Self {
+        LlmCommand {
+            prompt,
+            openai_client,
+        }
+    }
+
+    pub async fn generate_response(&self, input: Option<String>) -> Result<String, Box<dyn Error>> {
+        let context = if let Some(input) = input {
+            format!("{}: {}", self.prompt, input)
+        } else {
+            self.prompt.clone()
+        };
+        let output = self.openai_client.generate_text(&context, 100).await?;
+        Ok(output)
+    }
+}
+
+impl Runnable for LlmCommand {
+    fn run(&self) -> Result<String, Box<dyn Error>> {
+        let runtime = Runtime::new().unwrap();
+        let output = runtime.block_on(self.generate_response(None))?;
+        Ok(output)
+    }
+}
+
+impl ShellCommand for LlmCommand {
+    fn cmd(&self) -> &str {
+        "llm"
+    }
+
+    fn args(&self) -> Vec<&str> {
+        vec![&self.prompt]
+    }
+
+    fn pipe(&self, stdin: Option<Stdio>) -> Result<Option<ChildStdout>, Box<dyn Error>> {
+        let (pipe_out_r, pipe_out_w) = pipe()?;
+        let (pipe_err_r, pipe_err_w) = pipe()?;
+
+        match unsafe { fork()? } {
+            ForkResult::Parent { child: _ } => {
+                drop(pipe_out_w);
+                drop(pipe_err_w);
+                Ok(Some(ChildStdout::from(pipe_out_r)))
+            }
+            ForkResult::Child => {
+                drop(pipe_out_r);
+                drop(pipe_err_r);
+                dup2(pipe_out_w.as_raw_fd(), 1)?;
+                dup2(pipe_err_w.as_raw_fd(), 2)?;
+
+                let output = if let Some(stdin) = stdin {
+                    let mut child = Command::new("cat")
+                        .stdin(stdin)
+                        .stdout(Stdio::piped())
+                        .spawn()?;
+
+                    let stdout = child.stdout.take().ok_or("Failed to capture stdout")?;
+
+                    let mut reader = BufReader::new(stdout);
+                    let mut input = String::new();
+                    reader.read_to_string(&mut input)?;
+
+                    let runtime = Runtime::new().unwrap();
+                    runtime.block_on(self.generate_response(Some(input)))?
+                } else {
+                    let runtime = Runtime::new().unwrap();
+                    runtime.block_on(self.generate_response(None))?
+                };
+
+                // Write the output to stdout, which is piped
+                println!("{}", output);
+
+                std::process::exit(0);
+            }
+        }
```

Comment:
Why do you need to fork processes here?

Could you not wait for the response, then cat it like a normal `ExternalCommand`?